### ISSUE TITLE
ILookup<,> does not round trip properly for missing keys

### ISSUE THREAD

**Issue by drasive** [Cysharp/MemoryPack#171]
**The same issue as in MessagePack: https://github.com/MessagePack-CSharp/MessagePack-CSharp/issues/1431**

MemoryPack version used: 1.9.16

### Bug description
An ILookup<,> deserialized by MemoryPack throws a KeyNotFoundException when the provided key does not exist.

### Reproduction steps
```
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using MemoryPack;
using Xunit;

namespace Test
{
    public class MemoryPackTest
    {
        [Fact]
        public async Task Test()
        {
            var cacheFilePath = @"F:\Temp\cache_test";
            var lookup = new Dictionary<Guid, Guid>().ToLookup(x => x.Key, x => x.Value);

            _ = lookup[new Guid("bf5f1752-860c-46bb-98fe-ef440331cf90")]; // Does not throw an execption, good

            using (var writeStream = new FileStream(cacheFilePath, FileMode.Create))
            {
                await MemoryPackSerializer.SerializeAsync(writeStream, new DataWrapperTest() { Data = lookup });
            }

            using (var readStream = File.OpenRead(cacheFilePath))
            {
                var lookupWrapper = await MemoryPackSerializer.DeserializeAsync<DataWrapperTest>(readStream);
                var lookupCached = lookupWrapper.Data;

                _ = lookupCached[new Guid("bf5f1752-860c-46bb-98fe-ef440331cf90")]; // Throws KeyNotFoundException, bad!
            }
        }
    }

    [MemoryPackable]
    public partial class DataWrapperTest
    {
        public ILookup<Guid, Guid> Data { get; set; }
    }
}
```

### Expected behavior
No exception is thrown when accessing a deserialized ILookup<,> with a key that it does not contain.

### Actual behaviour
A KeyNotFoundException is thrown when accessing a deserialized  ILookup<,> with a key that it does not contain.

### Additional context
The workaround described in the [MessagePack issue](https://github.com/MessagePack-CSharp/MessagePack-CSharp/issues/1431) also applies.  
The fix applied in MessagePack may provide ideas for a fix in MemoryPack.

**Comment by neuecc** [comment]
Thank you, I see, I understand.
I would like to make a similar fix accordingly.

**Comment by hadashiA** [comment]
Fixed in #231
